protostar/protostar_toml/io/protostar_toml_reader.py

```diff
@@ -46,15 +45,31 @@ def get_attribute(
         if not section:
             return None
 
-        if attribute_name not in section:
-            attribute_name = attribute_name.replace("-", "_")
-            if attribute_name not in section:
-                return None
+        alternative_attribute_name = self._find_alternative_key(attribute_name, section)
+
+        if alternative_attribute_name and alternative_attribute_name in section:
+            return section[alternative_attribute_name]
+        return None
+
+    # pylint: disable=no-self-use
+    def _find_alternative_key(
+        self, base_key: str, raw_dict: Dict[str, Any]
+    ) -> Optional[str]:
+        if base_key in raw_dict:
+            return base_key
+
+        underscored_variant = base_key.replace("-", "_")
```

Comment:
Won't one of this be deprecated soon? I don't rememeber

Reply:
If it will please add a comment about it. Also maybe a warning?

Reply:
We can add a warning or support both syntaxes. IMO, it doesn't matter.

Reply:
I'll do it in one of the next PRs, because this warning is the responsibility of `ArgumentValueFromConfigProvider`.